shared/src/question-generators/graph-algorithms/dijkstraTable.ts

```diff
@@ -0,0 +1,440 @@
+  },
+}
+
+export const DijkstraTableGenerator: QuestionGenerator = {
+  id: "dijktable",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["graph", "dijkstra", "table"],
+  languages: ["en", "de"],
+  expectedParameters: [
+    {
+      name: "size",
+      type: "integer",
+      min: 2,
+      max: 8,
+    },
+  ],
+
+  generate: (lang: Language, parameters, seed) => {
+    const random = new Random(seed)
+
+    const { graph, steps } = generateRandomGraphWithSteps(random, parameters.size as number)
+
+    const permalink = serializeGeneratorCall({
+      generator: DijkstraTableGenerator,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const table = getDijkstraInputTable(steps, graph)
+    const feedback = getFeedbackFunction(steps, graph)
+    const checkFormat = getCheckFormatFunction(lang, graph)
+
+    console.log("debug:\n", cheatForDebugging(steps, graph))
+
+    const question: MultiFreeTextQuestion = {
+      type: "MultiFreeTextQuestion",
+      path: permalink,
+      name: t(translations, lang, "name"),
+      fillOutAll: false,
+      text: `${t(translations, lang, "tableQuestion")}\n${graph.toMarkdown()}\n${t(
+        translations,
+        lang,
+        "tablePrompt",
+      )}\n\n${table}`,
+      feedback,
+      checkFormat,
+    }
+
+    return { question }
+  },
+}
+
+function cheatForDebugging(
+  steps: Array<{
+    set: Set<string>
+    distances: Record<string, string>
+    predecessors: Record<string, string>
+  }>,
+  graph: Graph,
+): string {
+  return steps
+    .map((step) => {
+      const expectedSet = `{${[...step.set].join(", ")}}`
+      const nodeFeedback = graph.nodes
+        .map((node) => {
+          const nodeLabel = node.label!
+          const expectedDistance = step.distances[nodeLabel] || ""
+          const expectedPredecessor = step.predecessors[nodeLabel] || ""
+          return `${expectedDistance} | ${expectedPredecessor}`
+        })
+        .join(" | ")
+
+      return `| ${expectedSet} | ${nodeFeedback} |`
+    })
+    .join("\n")
+}
+
+interface DijkstraResult {
+  graph: Graph
+  steps: Array<{
+    set: Set<string>
+    distances: Record<string, string>
+    predecessors: Record<string, string>
+  }>
+}
+
+function generateRandomGraphWithSteps(random: Random, size: number): DijkstraResult {
+  const maxRetries = 100
+  let retries = 0
+  let edgeChance = 0.6
+
+  // guarantee more than one row in the algorithm
+  while (retries < maxRetries) {
+    const graph = createRandomGraph(random, size, edgeChance)
+    const steps = getDijkstraStepsIfValid(graph)
+    if (steps && steps.length > 1) {
+      return { graph, steps }
+    }
+
+    // increment retries and adjust edgeChance for next attempt
+    retries++
+    edgeChance += 0.1 // make graph denser on retry
+    console.log("retries so far:", retries)
+  }
```

Comment:
After 4 tries the prob that an edge exists is 1. After that 1.1 and so on. But does that change anything? 
Maybe instead of increasing the edgeChange always by 0.1 we can also increase the size?
Something like `(edgeChange, size) -> (0.6, size) (0.7, size) (0.8, size) (0.9, size) (0.6, size + 1) (0.7, size + 1)`. 

Or do I overlook soemthing?